src/kernel/cs_main.h

```diff
@@ -17,6 +17,6 @@
  * The transaction pool has a separate lock to allow reading from it and the
  * chainstate at the same time.
  */
-extern RecursiveMutex cs_main;
+inline RecursiveMutex cs_main;
```

Comment:
I checked if this could also be a problem for other extern symbols that are currently not used in `bitcoin-chainstate`, but still part of the kernel library. For example in `src/validation.h` we have `extern const std::vector<std::string> CHECKLEVEL_DOC;`, and it looks like using it is indeed problematic too if it were used externally: https://github.com/TheCharlatan/bitcoin/actions/runs/11531552497/job/32102377337#step:10:425.
There are a bunch of other such cases, where I'm not entirely sure how to handle them. Could just ignore them for now, but it is a bit annoying that the library might contain symbols that you couldn't just use.